platform/android/src/asset_manager_file_source.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <mbgl/storage/file_source.hpp>
+
+#include "asset_manager.hpp"
+
+#include <jni/jni.hpp>
+
+namespace mbgl {
+
+namespace util {
+template <typename T> class Thread;
+} // namespace util
+
+class AssetManagerFileSource : public FileSource {
+public:
+    AssetManagerFileSource(jni::JNIEnv&, jni::Object<android::AssetManager>);
+    ~AssetManagerFileSource() override;
+
+    std::unique_ptr<AsyncRequest> request(const Resource&, Callback) override;
+
+private:
+    jni::UniqueObject<android::AssetManager> assetManager;
+    class Impl;
+    std::unique_ptr<util::Thread<Impl>> thread;
```

Comment:
We do, but I've modeled this FileSource after all the other FileSources, and they are still using `util::Thread`. I attempted to port them to the actor system in https://github.com/mapbox/mapbox-gl-native/pull/7678